web-common/src/features/editor/Editor.svelte

```diff
@@ -34,6 +46,30 @@
     });
   });
 
+  function updateEditorExtensions(newExtensions: Extension[]) {
+    editor.setState(
+      EditorState.create({
+        doc: blob,
+        extensions: [
+          // establish a basic editor
+          base(),
+          // any extensions passed as props
+          ...newExtensions,
+          EditorView.updateListener.of((v) => {
+            if (v.focusChanged && v.view.hasFocus) {
+              dispatch("receive-focus");
+            }
+            if (v.docChanged) {
+              latest = v.state.doc.toString();
+
+              if (autoSave) debounceSave();
```

Comment:
Add !hideAutoSave or !FILES_WITHOUT_AUTOSAVE conditional check here, perhaps?

Reply:
Yeah, that's simpler, good call